mlflow/entities/evaluation_dataset.py

```diff
@@ -315,6 +339,142 @@ def _infer_source_types(self, record_dicts: list[dict[str, Any]]) -> None:
                     "source_data": {},
                 }
 
+    def _normalize_multiturn_records(
+        self, record_dicts: list[dict[str, Any]]
+    ) -> list[dict[str, Any]]:
+        """
+        Normalize records by moving top-level persona, goal, context into inputs.
+
+        {"persona": "Student", "goal": "Find articles"} becomes
+        {"inputs": {"persona": "Student", "goal": "Find articles"}}
+
+        Args:
+            record_dicts: List of record dictionaries to normalize
+
+        Returns:
+            List of normalized record dictionaries with multiturn fields inside inputs
+
+        Raises:
+            MlflowException: If record has both 'inputs' and top-level multiturn fields,
+                or contains unknown columns in multiturn format
+        """
+        normalized = []
+
+        for record in record_dicts:
+            if not (set(record.keys()) & MULTITURN_INPUT_FIELDS):
+                normalized.append(record.copy())
+                continue
+
+            if "inputs" in record:
+                raise MlflowException.invalid_parameter_value(
+                    "Cannot specify both 'inputs' and top-level multiturn fields "
+                    "(persona, goal, context). Use one or the other."
+                )
+
+            if unknown_columns := set(record.keys()) - MULTITURN_ALLOWED_COLUMNS:
+                raise MlflowException.invalid_parameter_value(
+                    f"Unknown columns in multiturn format: {list(unknown_columns)}. "
+                    "Custom fields should be placed inside 'context'."
+                )
+
+            new_record = {"inputs": {}}
+            for key, value in record.items():
+                if key in MULTITURN_INPUT_FIELDS:
+                    new_record["inputs"][key] = value
+                else:
+                    new_record[key] = value
+            normalized.append(new_record)
+
+        return normalized
+
+    def _validate_schema(self, record_dicts: list[dict[str, Any]]) -> None:
+        """
+        Validate schema consistency of new records and compatibility with existing dataset.
+
+        Args:
+            record_dicts: List of normalized record dictionaries
+
+        Raises:
+            MlflowException: If records mix multiturn/custom fields, use different schemas,
+                or are incompatible with existing dataset schema
+        """
+        batch_schema_type = None
+
+        for i, record in enumerate(record_dicts):
+            input_keys = set(record.get("inputs", {}).keys())
+
+            if not input_keys:
+                continue
+
+            record_type = self._classify_input_fields(input_keys)
+
+            if record_type == DatasetSchemaType.UNKNOWN:
+                custom_fields = input_keys - MULTITURN_INPUT_FIELDS
```

Comment:
can this be `UNKNOWN` for reasons other than multiturn? maybe we should make the error message more generic?